### DISCUSSION TITLE
How to display decimal/fractional values with uiSlider ? (i.e : 125.7)

### DISCUSSION THREAD

**Discussion by damiensellier** [RomanKubiak/ctrlr#257] [Q&A] [answered]
Hello,

My Yamaha SPX990 project is on the right way but I have some trouble to display decimal values with uiSliders.

For exemple, from the sysex message I get a decoded parameter value at 2554, that actually stands for 255.4 on the machine display (255.4 sec).

To update the modulator "MainFxParam15" I went with something like this:

```

	MainFxParamValue42 = SysexDecodedMessage:getByte(58)
	MainFxParamValue43 = SysexDecodedMessage:getByte(59)

	MainFxParam15Hold = (MainFxParamValue42*256 + MainFxParamValue43)
	console("Displays MainFxParam15Hold : "..(MainFxParam15Hold)) -- MainFxParam15Hold

	MainFxParam15 = panel:getModulatorByName("MainFxParam15")
	MainFxParam15:getComponent():setProperty("componentVisibility", 1, false)
	MainFxParam15:getComponent():setProperty("componentVisibleName", "Hold", true)
	MainFxParam15:getComponent():setProperty("uiSliderMin", 0, true)
	MainFxParam15:getComponent():setProperty("uiSliderMax", 24000, true)
	MainFxParam15:getComponent():setValue(MainFxParam15Hold/10, true)
	MainFxParam15:setProperty("uiSliderInterval","0.1", true)
	MainFxParam15:setProperty("modulatorCustomIndex","15", true)
	MainFxParam15:setProperty("modulatorCustomName","", true)
	MainFxParam15:getComponent():setProperty("uiType","uiSlider", true)
```

so my value is actually :  MainFxParam15Hold / 10  for 2554 / 10 = 255.4
But the value is automatically rounded to 255 even with uiSliderInterval set at 0.1
I will mulitply the value back to x10 to have the proper sent message value later on.

How do you usually display numbers with decimal values in your modulator?

Thanks for your help!

PS: updating uiType slider to fixedslider with lua "	**mod:getComponent():setProperty("uiType","uiSlider", true)**"  is really neat!

**Comment by DRBAUS** [comment, marked as answer]
Is using a uiLabel an option for you to display values instead of using the Modulator Value box?

**Comment by dobo365** [comment]
Hi Damien,
  indeed, as @DRBAUS mentions you should use a separate uiLabel to display the values. I'm myself never using the built-in value display.
I'm doing that in all my panels and I'm handling all similar modulators with a magic formula in one single method.
Always keeping the uiSliderInterval to 1 but then, deciding on the Max value according to the precision needed.
What is the precision on your SPX? Can you move values by 0.1? From where to where? If 0 to 300 then just set Max at 3000 and use a calculation to display the right value with the decimal.

**Reply by damiensellier** [reply]
Hello, this is exactly what I'm going to do. I was wondering if there was a way to keep the decimals but nevermind I'll update the main uiLabel with the value ÷ 10 and that's it. I'll keep the raw value on the modulator.
I had another idea in mind maybe use a uiFixedSlider and put all the values from 0.0 to 400.0 with 0.1 increments but it requires a kind of text generator that allows to separate values with \n inbetween and optionally append/prepend for prefix/suffixes.
 Do you know a text tool that could that? that would be nice even for generating "slider content" for uiFixedSlider like +1dB\n+2dB\n+3dB\n+4dB etc

Thanks guys I'll go with the label update.

**Reply by Ted-juh** [reply]
https://pinetools.com/generate-list-numbers

n+1dB
n+2dB
n+3dB
n+4dB
n+5dB
n+6dB
n+7dB
n+8dB
n+9dB
n+10dB

**Reply by damiensellier** [reply]
Thanks @DnaldeT this is the tool I was looking for. 100% sure I'll need that someday :)

**Reply by Ted-juh** [reply]
Or put a for loop in combination with a string.format at some button. That way you can create a numbered list as well.

for i = 1,10 do
list = string.format("n+%sdB", i)
console(""..list)
end
 
The ("") might not be necessary.
And copy it from the console output.

**Reply by dobo365** [reply]
Or use Excel then copy/paste in Notepad
Anyway, I don't think that putting values with a regular increment in a table is a good idea or good programming practice...

**Reply by Ted-juh** [reply]
I think I'm missing something. Table?

**Reply by damiensellier** [reply]
It can be useful to generate lists for slider content
If i update a modulator for 1 parameter to another, I can't use table in this case : 

```
	MainFxParam13 = panel:getModulatorByName("MainFxParam13")
	MainFxParam13:getComponent():setProperty("componentVisibility", 1, false)
	MainFxParam13:getComponent():setProperty("componentVisibleName", "FbHigh", true)
	MainFxParam13:getComponent():setProperty("uiSliderMin", 1, true)
	MainFxParam13:getComponent():setProperty("uiSliderMax", 10, true)
	MainFxParam13:getComponent():setProperty("uiSliderDoubleClickEnabled", 1, true)
	MainFxParam13:getComponent():setProperty("uiSliderDoubleClickValue", 0, true)
	MainFxParam13:getComponent():setProperty("uiFixedSliderContent","0.1\n0.2\n0.3\n0.4\n0.5\n0.6\n0.7\n0.8\n0.9\n1.0", true)
	MainFxParam13:getComponent():setValue(MainFxParamValue45-1, true) -- starts at 0.1
	MainFxParam13:setProperty("modulatorCustomIndex","13", true)
	MainFxParam13:setProperty("modulatorCustomName","", true)
	MainFxParam13:getComponent():setProperty("uiType","uiFixedSlider", true)
```

This way I can populate with a list of values for the fixedslider
`MainFxParam13:getComponent():setProperty("uiFixedSliderContent","0.1\n0.2\n0.3\n0.4\n0.5\n0.6\n0.7\n0.8\n0.9\n1.0", true)`
